xbout/wall.py

```diff
@@ -0,0 +1,119 @@
+"""
+Routines to read and represent wall geometries
+"""
+
+import numpy as np
+
+
+class AxisymmetricWall:
+    def __init__(self, Rs, Zs):
+        """
+        Defines a 2D (R,Z) axisymmetric wall
+
+        Parameters
+        ----------
+
+        Rs : list or 1D array
+            Major radius coordinates [meters]
+        Zs : list or 1D array
+            Vertical coordinates [meters]
+
+        """
+        if len(Rs) != len(Zs):
+            raise ValueError("Rs and Zs arrays have different lengths")
+
+        # Ensure that the members are numpy arrays
+        self.Rs = np.array(Rs)
+        self.Zs = np.array(Zs)
+
+    def __iter__(self):
+        """
+        Iterate over wall elements
+
+        Each iteration returns a pair of (R,Z) pairs:
+            ((R1, Z1), (R2, Z2))
+
+        These pairs define wall segment.
+        """
+        return iter(
+            zip(zip(self.Rs, self.Zs), zip(np.roll(self.Rs, -1), np.roll(self.Zs, -1)))
+        )
+
+    def to_polygon(self):
+        """
+        Returns a 2D Numpy array [npoints, 2]
+        Index 0 is major radius (R) in meters
+        Index 1 is height (Z) in meters
+        """
+        return np.stack((self.Rs, self.Zs), axis=-1)
+
+    def plot(self, linestyle="k-", ax=None):
```

Comment:
FYI I think plotting the wall onto the plasma is analogous to plotting features like coastlines using cartopy onto the earth.

There might be useful inspiration from how cartopy interacts with matplotlib, especially if you want to set up presets for a small number of possible wall shapes (there are only so many tokamaks in the world).